lib/src/main/java/uk/gov/di/ipv/cri/passport/library/service/AccessTokenService.java

```diff
@@ -58,11 +59,23 @@ public String getResourceIdByAccessToken(String accessToken) {
         return Objects.isNull(accessTokenItem) ? null : accessTokenItem.getResourceId();
     }
 
-    public void persistAccessToken(AccessTokenResponse tokenResponse, String resourceId) {
+    public void persistAccessToken(
+            AccessTokenResponse tokenResponse, String resourceId, String authCode) {
         AccessTokenItem accessTokenItem = new AccessTokenItem();
         accessTokenItem.setAccessToken(
                 DigestUtils.sha256Hex(tokenResponse.getTokens().getBearerAccessToken().getValue()));
         accessTokenItem.setResourceId(resourceId);
+        accessTokenItem.setAuthCode(authCode);
+
         dataStore.create(accessTokenItem);
     }
+
+    public void revokeAccessTokenViaAuthCode(String authCode) throws IllegalArgumentException {
+        AccessTokenItem accessTokenItem =
+                ListUtil.getOneItemOrThrowError(
+                        dataStore.getItemByIndex(
+                                "authorizationCode-index", DigestUtils.sha256Hex(authCode)));
+
+        dataStore.delete(DigestUtils.sha256Hex(accessTokenItem.getAccessToken()));
```

Comment:
I don't think you want the `DigestUtils.sha256Hex()` in here. The access token item was already stored with it's token hashed, so you'd be hashing twice here.